store/src/main/java/org/apache/rocketmq/store/schedule/ScheduleMessageService.java

```diff
@@ -317,7 +318,7 @@ public void executeOnTimeup() {
                         long nextOffset = offset;
                         int i = 0;
                         ConsumeQueueExt.CqExtUnit cqExtUnit = new ConsumeQueueExt.CqExtUnit();
-                        for (; i < bufferCQ.getSize(); i += ConsumeQueue.CQ_STORE_UNIT_SIZE) {
+                        for (; i < bufferCQ.getSize() && isStarted(); i += ConsumeQueue.CQ_STORE_UNIT_SIZE) {
```

Comment:
`executeOnTimeup()` method will only invoke in `run()` method, which already has the condition of `isStarted()`, so why add a `isStarted()` here
https://github.com/apache/rocketmq/pull/3478/files#diff-350d772aa64ef5e5f5012863595bc2e9d35c731fbba75a81b823cf00b1a2d593R281-R282

Reply:
When we close the service, we cannot guarantee that it has been executed in this for loop.

Reply:
I see